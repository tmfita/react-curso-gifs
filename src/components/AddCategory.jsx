import { useState } from "react"
import PropTypes from 'prop-types'


export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const onInputChange = (event) => {
    setInputValue(event.target.value)
  }
  const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length < 1) return;
    //setCategories(cat => [...cat, inputValue])
    onNewCategory( inputValue.trim() )
    setInputValue('');
  }
  return (
    <form onSubmit={ onSubmit } aria-label="form">

      <input 
        type="text" 
        placeholder="Buscar gif"
        value = { inputValue }
        onChange = { onInputChange }
      />

    </form>
  )
}
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}