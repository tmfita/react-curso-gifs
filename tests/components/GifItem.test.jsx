/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas de GifItem', () => { 
    const title = "Soy un título";
    const url = "https://somcomunitatenergetica.cat/static/img/somComLogo.52c2aa7.svg";
    test('GifItem debe ser igual al snapshot', () => {
       const { container } =  render( <GifItem title={ title } url={ url } />);
       expect( container ).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el url y el alt indicado', () => {
        render( <GifItem title={ title } url={ url } /> )
        screen.debug();
        expect( screen.getByRole( 'img' ).src ).toBe( url )
        expect( screen.getByRole( 'img' ).alt ).toBe( title )
        const { src, alt } = screen.getByRole( 'img' )
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
    });

    test('debe mostrar el título en el componente', () => {
        render( <GifItem title={ title } url={ url } /> )
        expect( screen.getByText( title )).toBeTruthy()
    })
 })