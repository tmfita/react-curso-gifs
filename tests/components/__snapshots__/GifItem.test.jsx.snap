// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem GifItem debe ser igual al snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un título"
      src="https://somcomunitatenergetica.cat/static/img/somComLogo.52c2aa7.svg"
    />
    <p>
      Soy un título
    </p>
  </div>
</div>
`;
